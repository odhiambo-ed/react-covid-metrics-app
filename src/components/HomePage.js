import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { addCountries } from '../features/countriesReducer';

import '../App.css';
import CountrySelection from './CountrySelection';
import Naigation from './Navigation';

const HomePage = () => {
  const [search, setSearch] = useState('');
  const dispatch = useDispatch();
  const countries = useSelector((state) => state.countries.value);

  useEffect(() => {
    fetch('https://api.covid19tracking.narrativa.com/api/countries')
      .then((res) => res.json())
      .then((final) => {
        dispatch(addCountries(final.countries));
      });
  }, []);
  return (
    <div className="App">
      <Naigation />
      <div className="m-3 ">
        <input
          type="email"
          className="form-control form"
          id="search"
          value={search}
          onChange={(e) => setSearch(e.target.value)}
          placeholder="Enter country name to search..."
        />
      </div>
      {search.length > 0 ? (
        <div className="content">
          {countries
            .filter((val) => {
              if (val.name.toLowerCase().includes(search.toLowerCase())) {
                return val;
              }
              return '';
            })
            .map((val) => (
              <CountrySelection key={val.id} id={val.id} name={val.name} />
            ))}
        </div>
      ) : (
        <div className="content">
          {countries.map((item) => (
            <CountrySelection key={item.id} name={item.name} />
          ))}
        </div>
      )}
    </div>
  );
};

export default HomePage;
