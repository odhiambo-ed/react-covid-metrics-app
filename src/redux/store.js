import { configureStore } from '@reduxjs/toolkit';
import covidReducer from '../features/covidReducer';
import countriesReducer from '../features/countriesReducer';
import selectedCountryReducer from '../features/selectedCountryReducer';
import regionsReducer from '../features/regionsReducer';
import selectedRegionReducer from '../features/selectedRegionReducer';
import dateReducer from '../features/dateReducer';

const store = configureStore({
  reducer: {
    covid: covidReducer,
    countries: countriesReducer,
    country: selectedCountryReducer,
    regions: regionsReducer,
    region: selectedRegionReducer,
    date: dateReducer,
  },
});

export default store;
