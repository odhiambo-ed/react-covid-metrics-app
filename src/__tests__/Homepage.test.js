import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import NavBar from '../components/Navigation';
import HomePage from '../components/HomePage';

describe('Render components', () => {
  test('render NavBar', () => {
    const tree = renderer
      .create(<Router><Provider store={store}><NavBar /></Provider></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('render NavBar', () => {
    const tree = renderer
      .create(<Router><Provider store={store}><HomePage /></Provider></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
