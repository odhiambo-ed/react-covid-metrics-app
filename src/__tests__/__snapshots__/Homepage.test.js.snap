// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render components render NavBar 1`] = `
<nav
  className="navbar navbar-light nav-c"
>
  <div
    className="container-fluid"
  >
    <img
      alt=""
      className="d-inline-block align-text-top"
      height="50"
      src="https://upload.wikimedia.org/wikipedia/commons/thumb/8/82/SARS-CoV-2_without_background.png/1193px-SARS-CoV-2_without_background.png?20210922032455"
      style={
        Object {
          "borderRadius": 999,
        }
      }
      width="50"
    />
    <div>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="MicIcon"
        focusable="false"
        style={
          Object {
            "color": "white",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M12 14c1.66 0 2.99-1.34 2.99-3L15 5c0-1.66-1.34-3-3-3S9 3.34 9 5v6c0 1.66 1.34 3 3 3zm5.3-3c0 3-2.54 5.1-5.3 5.1S6.7 14 6.7 11H5c0 3.41 2.72 6.23 6 6.72V21h2v-3.28c3.28-.48 6-3.3 6-6.72h-1.7z"
        />
      </svg>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="SettingsIcon"
        focusable="false"
        style={
          Object {
            "color": "white",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
        />
      </svg>
    </div>
  </div>
</nav>
`;

exports[`Render components render NavBar 2`] = `
<div
  className="App"
>
  <nav
    className="navbar navbar-light nav-c"
  >
    <div
      className="container-fluid"
    >
      <img
        alt=""
        className="d-inline-block align-text-top"
        height="50"
        src="https://upload.wikimedia.org/wikipedia/commons/thumb/8/82/SARS-CoV-2_without_background.png/1193px-SARS-CoV-2_without_background.png?20210922032455"
        style={
          Object {
            "borderRadius": 999,
          }
        }
        width="50"
      />
      <div>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MicIcon"
          focusable="false"
          style={
            Object {
              "color": "white",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M12 14c1.66 0 2.99-1.34 2.99-3L15 5c0-1.66-1.34-3-3-3S9 3.34 9 5v6c0 1.66 1.34 3 3 3zm5.3-3c0 3-2.54 5.1-5.3 5.1S6.7 14 6.7 11H5c0 3.41 2.72 6.23 6 6.72V21h2v-3.28c3.28-.48 6-3.3 6-6.72h-1.7z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="SettingsIcon"
          focusable="false"
          style={
            Object {
              "color": "white",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
          />
        </svg>
      </div>
    </div>
  </nav>
  <div
    className="m-3 "
  >
    <input
      className="form-control form"
      id="search"
      onChange={[Function]}
      placeholder="Enter country name to search..."
      type="email"
      value=""
    />
  </div>
  <div
    className="content"
  />
</div>
`;
